public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,823 @@
+      closeFlyout();
+    } catch (e) {
+    } finally {
+      formikHelpers.setSubmitting(false);
+    }
+  };
+
+  function formikValueToDetector(values): Detector {
+    const detectionDateRange = values.historical
+      ? {
+          startTime: convertTimestampToNumber(values.startTime),
+          endTime: convertTimestampToNumber(values.endTime),
+        }
+      : undefined;
+    var resultIndex = values.resultIndex;
+    if (resultIndex && resultIndex.trim().length > 0) {
+      resultIndex = CUSTOM_AD_RESULT_INDEX_PREFIX + resultIndex;
+    }
+    let detectorBody = {
+      name: values.name,
+      description: values.description,
+      indices: values.indices,
+      resultIndex: resultIndex,
+      filterQuery: formikToFilterQuery(values),
+      uiMetadata: {
+        features: {
+          ...featuresToUIMetadata(featureListToUIMetadata(values.featureList)),
+        },
+        filters: get(values, 'filters', []),
+      },
+      featureAttributes: formikToFeatureAttributes(values.featureList),
+      timeField: values.timeField,
+      detectionInterval: values.detectionInterval,
+      windowDelay: values.windowDelay,
+      shingleSize: values.shingleSize,
+      categoryField: !isEmpty(values?.categoryField)
+        ? values.categoryField
+        : undefined,
+    } as Detector;
+    // Optionally add detection date range
+    if (detectionDateRange) {
+      detectorBody = {
+        ...detectorBody,
+        //@ts-ignore
+        detectionDateRange: detectionDateRange,
+      };
+    }
+
+    return detectorBody;
+  }
+
+  function featureListToUIMetadata(featureList) {
+    return featureList.map(function (value) {
+      const oneFeature = {
+        featureId: value.featureId,
+        featureName: value.featureName,
+        featureType: value.featureType,
+        featureEnable: true,
+        importance: 1,
+        aggregationBy: value.aggregationBy,
+        aggregationQuery:
+          value.featureType === 'simple_aggs'
+            ? '{}'
+            : formikToUIMetadataAggregation(
+                value.featureId,
+                value.aggregationBy,
+                value.aggregationOf[0].label
+              ),
+        newFeature: true,
+        aggregationOf: formikToAggregationOf(value),
+      };
+
+      return oneFeature;
+    });
+  }
+
+  const initialDetectorValue = {
+    name: detectorNameFromVis,
+    indices: [embeddable.vis.data.aggs.indexPattern.title],
+    timeField: embeddable.vis.data.indexPattern.timeFieldName,
+    detectionInterval: {
+      period: { interval: intervalValue, unit: UNITS.MINUTES },
+    },
+    windowDelay: {
+      period: { interval: delayValue, unit: UNITS.MINUTES },
+    },
+    shingleSize: shingleSizeValue,
+    filterQuery: { match_all: {} },
+    description: '',
+    resultIndex: undefined,
+    filters: [],
+    featureList: featureListToFormik(featureList),
+    categoryFieldEnabled: false,
+    realTime: true,
+    historical: false,
+  };
```

Comment:
Some of these values we can use the existing default values for. For example, initial formik values on the existing creation flow are here:
detector definition values: [link](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/DefineDetector/utils/constants.tsx#L36)
model configuration values: [link](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/ConfigureModel/utils/constants.tsx#L19)

Reply:
Sure, no hard opinion on that, makes sense to me. May be more difficult to try to pull those defaults in, given we need to set so many here already based on the vis. I'm good with that